File: CS/WindowsFormsApplication475/Form1.cs
using DevExpress.Utils;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication475 {
    public partial class Form1 : Form {
        RepositoryItemCustomEdit edit;
        public Form1() {
            InitializeComponent();
            edit = new RepositoryItemCustomEdit();
            List<Employee> lst = new List<Employee>();
            for (int i = 0; i < 10; i++) {
                lst.Add(new Employee() { ID = i, FirstName = "Name " + i, BirthDate = DateTime.Now, AtVacation = i % 2 == 0, FavoriteColor = Color.Red });
            }
            gridControl1.DataSource = lst;
            gridView1.Columns["FavoriteColor"].ColumnEdit = edit;
        }
    }


    public class Employee {
        public int ID { get; set; }
        public string FirstName { get; set; }
        public bool AtVacation { get; set; }
        public DateTime BirthDate { get; set; }
        public Color FavoriteColor { get; set; }
    }

}

File: VB/WindowsFormsApplication475/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsFormsApplication475
	Partial Public Class Form1
		Inherits Form
		Private edit As RepositoryItemCustomEdit
		Public Sub New()
			InitializeComponent()
			edit = New RepositoryItemCustomEdit()
			Dim lst As New List(Of Employee)()
			For i As Integer = 0 To 9
				lst.Add(New Employee() With {.ID = i, .FirstName = "Name " & i, .BirthDate = DateTime.Now, .AtVacation = i Mod 2 = 0, .FavoriteColor = Color.Red})
			Next i
			gridControl1.DataSource = lst
			gridView1.Columns("FavoriteColor").ColumnEdit = edit
		End Sub
	End Class


	Public Class Employee
		Private privateID As Integer
		Public Property ID() As Integer
			Get
				Return privateID
			End Get
			Set(ByVal value As Integer)
				privateID = value
			End Set
		End Property
		Private privateFirstName As String
		Public Property FirstName() As String
			Get
				Return privateFirstName
			End Get
			Set(ByVal value As String)
				privateFirstName = value
			End Set
		End Property
		Private privateAtVacation As Boolean
		Public Property AtVacation() As Boolean
			Get
				Return privateAtVacation
			End Get
			Set(ByVal value As Boolean)
				privateAtVacation = value
			End Set
		End Property
		Private privateBirthDate As DateTime
		Public Property BirthDate() As DateTime
			Get
				Return privateBirthDate
			End Get
			Set(ByVal value As DateTime)
				privateBirthDate = value
			End Set
		End Property
		Private privateFavoriteColor As Color
		Public Property FavoriteColor() As Color
			Get
				Return privateFavoriteColor
			End Get
			Set(ByVal value As Color)
				privateFavoriteColor = value
			End Set
		End Property
	End Class

End Namespace
